app/src/main/java/com/amaze/filemanager/ui/views/drawer/CustomNavigationView.java

```diff
@@ -91,6 +104,20 @@ public void onRestoreInstanceState(Parcelable state) {
         this.checkedId = ss.selectedId;
     }
 
+    /**
+     * This is a hack, when the SavedState class is unmarshalled a "ClassNotFoundException"
+     * will be thrown (the actual class not found is "android.support.design.widget.NavigationView$SavedState")
+     * and I seem to only be able to replicate on Marshmallow.
+     * Trying to find the class and returning false if Class.forName() throws "ClassNotFoundException"
+     * doesn't work because the class seems to have been loaded with the current loader
+     * (not the one the unmarshaller uses); of course I have no idea of what any of this means so
+     * I could be wrong.
+     * For the crash see https://github.com/TeamAmaze/AmazeFileManager/issues/1101.
+     */
+    public boolean doesNavigationView$SavedStateExist() {
```

Comment:
I have reservations on having dollar signs in method names, as such naming convention is usually used by generated code - e.g. `Foo$Bar.class`, `Foo$1.class`, or private methods available between inner class and its enclosing class - as read from [here](https://stackoverflow.com/questions/7578155/meanings-of-dollar-sign-in-java-method-descriptor).

While Dalvik is not strictly JVM (ala Hotspot), and it is perfectly legal syntax, I'd prevent using dollar signs in my Java code to prevent any problem that may be caused.

I prefer name it `isNavigationView_SavedStateExist()`.

Reply:
Done, but I despise '_' so I just removed the '$'.